### PR TITLE
5 get/api/articles/article

### PR DESCRIPTION
**Pull request by valy4**

### CODE REVIEW

File: Controllers/controllers.errors.js [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
 };
+exports.handleCustomPaths = (err, req, res, next) => {
+  if (err.msg !== undefined) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+};
+exports.handle400Paths = (err, req, res, next) => {
+  res.status(400).send({ msg: "Bad request - invalid id" });
```
Comment (Poonam-raj) [inline Controllers/controllers.errors.js]: Remember that you still need to be able to pass errors to the 500 error handler if they aren't 400 errors - you still need a If Else situation in handle400Paths

File: __tests__/app.test.js [hunk 2]
```diff
@@ -81,5 +81,43 @@ describe("4.GET /api/articles", () => {
         expect(articles).toBeSortedBy("created_at", { descending: true });
       });
   });
-  
+});
+describe.only("5.GET /api/articles/:article_id", () => {
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: Valentina please remove the .only off this test block, re-run your tests, re-commit the code and make another PR when you're done. We can only review code that has run all the tests before commiting

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Please remove .only and re-submit 🚀

File: Controllers/controllers.js [hunk 3]
```diff
@@ -9,13 +13,23 @@ exports.getTopics = (req, res, next) => {
...
\ No newline at end of file
+    .then((articles) => {
+      res.status(200).send({ articles });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+exports.getArticleById = (req, res, next) => {
+  const { article_id } = req.params;
```
Comment (Pipe-Smoking-Rabbit) [inline Controllers/controllers.js:27]: very nice deconstruction

File: Models/models.js [hunk 4]
```diff
@@ -20,3 +18,15 @@ exports.selectArticles = (sort_by = "created_at") => {
...
 };
+
+exports.selectArticleById = (article_id) => {
+  const SQL = "SELECT * FROM articles WHERE article_id = $1";
+
+  return db.query(SQL, [article_id]).then((result) => {
+    if (result.rowCount === 0) {
+      return Promise.reject({ msg: "Page not found!", status: 404 });
+    } else {
+      return result.rows;
```
Comment (Pipe-Smoking-Rabbit) [inline Models/models.js]: because this is just a single object that we're trying to get - you should return `result.rows[0]` to return the object not inside an array

File: __tests__/app.test.js [hunk 5]
```diff
@@ -81,5 +81,43 @@ describe("4.GET /api/articles", () => {
...
+describe("5.GET /api/articles/:article_id", () => {
+  test("responds with statusCode 200 and the response contains an article object ", () => {
+    return request(app)
+      .get("/api/articles/2")
+      .expect(200)
+      .then(({ body }) => {
+        const { article } = body;
+        expect(article).toHaveLength(1);
+        expect(article[0].article_id).toBe(2);
+        expect(article[0]).toEqual(
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: once you change the model to return `results.rows[0]` that will remove the need for the `[0]` in these tests - and also remove the need for the check of `.length`

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Looks really good! Just make a small quick change to the model so it's returning a single object not inside an array - check my comments for details - once that's done feel free to merge! :)
